sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.students.controller.updateApplicationOfferingChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,242 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+  getStudentByFakeStudentUserType,
+} from "../../../../testHelpers";
+import {
+  saveFakeApplicationOfferingRequestChange,
+  createE2EDataSources,
+  E2EDataSources,
+  saveFakeApplication,
+} from "@sims/test-utils";
+import {
+  ApplicationOfferingChangeRequestStatus,
+  ApplicationStatus,
+  Student,
+} from "@sims/sims-db";
+
+describe("ApplicationOfferingChangeRequestStudentsController(e2e)-updateApplicationOfferingChangeRequest", () => {
+  let app: INestApplication;
+  let student: Student;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    student = await getStudentByFakeStudentUserType(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+      dataSource,
+    );
+  });
+
+  it("Should update the application offering change request status for the provided application offering change request id.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(
+      db.dataSource,
+      {
+        student,
+      },
+      { applicationStatus: ApplicationStatus.Completed },
+    );
+    const applicationOfferingChangeRequest =
+      await saveFakeApplicationOfferingRequestChange(db, {
+        application,
+      });
+    const token = await getStudentToken(
```

Comment:
since token is the same for all TC, maybe we can move it to beforeAll. same for other files

Reply:
We usually avoid doing that to ensure that we will have a not expired token. For these TCs it seems to be ok but I believe that we should avoid it as a general practice.

Reply:
Okay, I got it. I got these comments in my previous PR, But now I remember why we have token in TC-level

Reply:
I have put the token inside `beforeEach`. I believe that's ok.